src/connectionbenchmark/benchmarktaskdownload.cpp

```diff
@@ -31,43 +40,80 @@ void BenchmarkTaskDownload::handleState(BenchmarkTask::State state) {
   logger.debug() << "Handle state" << state;
 
   if (state == BenchmarkTask::StateActive) {
-    m_request = NetworkRequest::createForGetUrl(this, m_fileUrl);
-    connect(m_request, &NetworkRequest::requestUpdated, this,
+    // Start DNS resolution
+#if !defined(MVPN_DUMMY)
+    m_dnsLookup.setNameserver(QHostAddress(MULLVAD_DEFAULT_DNS));
+#endif
+    m_dnsLookup.lookup();
+  } else if (state == BenchmarkTask::StateInactive) {
+    for (NetworkRequest* request : m_requests) {
```

Comment:
Is there a way to cancel the DNS lookup?

Reply:
Yes, there is `QDnsLookup::abort()` that ought to cancel. I'll add another commit to address it.